src/util/MemorySize/MemorySize.cpp

```diff
@@ -63,7 +58,39 @@ std::string MemorySize::asString() const {
 
 // _____________________________________________________________________________
 MemorySize MemorySize::parse(std::string_view str) {
-  return MemorySizeParser::parseMemorySize(str);
+  if (auto matcher =
+          ctre::match<"(\\d+(?:\\.\\d+)?) ([kKmMgGtT][bB]?|[bB])">(str)) {
+    auto amountString = matcher.get<1>().to_view();
+    double amount;
+    std::from_chars(amountString.begin(), amountString.end(), amount);
+    auto unitString = matcher.get<2>().to_view();
+    switch (unitString.at(0)) {
+      case 'b':
+      case 'B':
+        if (ad_utility::contains(amountString, '.')) {
+          break;
+        }
```

Comment:
Yes, but in that case I'd have to change some of the other logic too; If I write `1.2345 KB` I'M pretty sure it will just get cast to `1234 B`. So the main reason why I added this distinction was me wanting to make the unit test pass so the behaviour stays the same.

Reply:
I don't understand what this has to do with any `KB` strings.
The `break` here has the effect, that you get the error `1.304B could not be parsed....`.
And in this particular case I would like to add a note, that for bytes the decimal point is not allowed. Either throw directly instead of the break, or set a flag, s.t. you know that you have to append the note when throwing. I hope this doesn't break the unit test (Otherwise I hope you can change the corresponding matcher in a single position and add one for the explicit suffix for the special case.